src/test/java/in/org/projecteka/hdaf/link/clients/ClientRegistryClientTest.java

```diff
@@ -68,4 +70,31 @@ void getProvidersByGivenName() throws IOException {
 
         assertThat(captor.getValue().url().toString()).isEqualTo("localhost:8000/providers?name=Max");
     }
+
+    @Test
+    void getProviderByGivenId() throws IOException {
+        var source = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false).readValue(
```

Comment:
Do we really need to load the `provider.json`,  can't we just have Json string in the test case itself?